### PR TITLE
Add CodeCov to repo

### PR DESCRIPTION
**Pull request by imshashank**
Adding CodeCov to the repo.

This update deprecated phpunit 4 in favor of phpunit 4.8

Please review the code @kstich @mtdowling @jeskew @cjyclaire

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "ext-json": "*",
         "ext-dom": "*",
         "ext-simplexml": "*",
-        "phpunit/phpunit": "~4.0|~5.0",
+        "phpunit/phpunit": "~4.8",
```
Comment (jeskew) [inline composer.json]: 4.8 is pretty far out of date at this point and isn't guaranteed to be compatible with PHP 7 or 7.1. To make sure the tests are still compatible with all of the versions we test in Travis, I would recommend `^4.8.35|^5.4.0|^6.0.0`

Comment (jeskew) [inline composer.json]: Looks like the underscore-namespaced class names were dropped from PHPUnit 6. `^4.8.35|^5.4.0` should avoid the failures that were reported for PHP 7 and 7.1. Alternatively, you could migrate to the new test class naming convention (e.g., replace `PHPUnit_Framework_TestCase` with `\PHPUnit\Framework\TestCase`).

Comment (imshashank) [inline composer.json]: I'll add this task to backLog to replace PHPUnit_Framework_TestCase with \PHPUnit\Framework\TestCase.
Have updated the composer.json file with values: ^4.8.35|^5.4.0",

**Comment by imshashank** [comment]
@jeskew @kstich @mtdowling @cjyclaire Please let me know if this requires any more changes?

**Comment by kstich** [comment]
If we're okay with the changes to the required minimum versions of PHPUnit, this looks good to me.

**Pull request merged**